### DISCUSSION TITLE
Flutter Form Builder Ecosystem

### DISCUSSION THREAD

**Discussion by deandreamatias** [danvick/flutter_form_builder#1038] [General]
With the changes in version 6.0.0 in flutter_form_builder, which split the components to different packages and repositories, the understanding and use by developers became a bit more complicated.
The main cause of having to split the packages, was the migration to null safety which has already been done on all dependencies (it seems).

Now with the stable null safety, the split packages and the more active help from other contributors, I think it may be time to think about improvements for this ecosystem of packages that are so used and needed. 
I have thought of a few things to maintain and improve this ecosystem of open source form solutions made by and for the Flutter community.
Below I will list points that to me seem important but obviously can be discussed by everyone.

Objectives:

1. Facilitate the development of packages.
2. To be an open source package made by the whole community.
3. To be an easy to use package
4. To be a stable package to build forms

Road map:

- [x] 1.1. Analyze the possible integration of packages from other repositories into this one (such as https://github.com/danvick/form_builder_file_picker) even if they are distributed in different packages.
- [x] 1.2. If there are packages outside this repository but related to the ecosystem, link them and apply the same development and deployment strategies. Ex: Issue templates, readme, etc
- [ ] 1.3. Create a simple and clear branching strategy for deployment of a stable and a beta version.
- [x] 1.4. Add contributing
- [x]   1.4.1. How to open issues
- [x]   1.4.2. How to make pull requests
- [x]   1.4.3. Explanation of the different branches
- [X] 2.1 Add code of conduct
- [x] 2.2 Encourage more people to contribute
- [x]   2.2.1 Message in readme
- [ ]   2.2.2 Articles, videos, social media posts
- [x]   2.2.3 Post messages in communities around the world
- [x] 3.1 Document a list of the different packages and their available components.
- [x] 3.2 Update packages with the latest versions of dependencies and Flutter.
- [x] 3.3 Update examples
- [ ] 3.4 Add visual examples (images, gifs, videos, sample application)
- [x] 4.1 Review open issues and categorize them with labels
- [ ] 4.2 Resolve the different issues, prioritizing by bugs. Long term and continuous work.

I repeat, this list is my ideas but it is totally modifiable.
Obviously this is a work for a long time, but it will give long life to the package.

Thanks to all and specially to @danvick for giving me the opportunity to be part of this project

**Comment by deandreamatias** [comment]
What do you think @WilliamCunhaCardoso @awhitford @MHosssam @ahmednfwela @ShadyZekry ?

**Reply by WilliamCunhaCardoso** [reply]
I totally agree with the suggestions. And as our colleague @ShadyZekry mentioned, due to being a long time work, it would be good a channel to discuss in more details it, as such a Discord server.

I am very interested in how we can initiate such enhancements, and willing to reunite and discuss it more.

**Reply by deandreamatias** [reply]
I think that Slack is used a lot by business but to open source communities, Discord will be better because has no limit messages like Slack.
Another thing is that discussions would to be totally open to community. For example, we can create voice channels in Discord but we need priorice Github to discuss and shared how we made the decisions

**Comment by ShadyZekry** [comment]
I think that's a very promising roadmap. But as you mentioned, Obviously this is a work for a long time.

We can start with discord server to organize these type of discussions there, accept other's  questions, make it easier and faster to get feedback about somethings like readme and code of conduct and other things. 
That was the first thing that came into my mind.

**Comment by deandreamatias** [comment]
I created a PR (https://github.com/danvick/flutter_form_builder/pull/1041) with contributing and code of conduct base files. We can start from there.
Another PR is to update flutter_form_phone_field https://github.com/danvick/flutter_form_builder/pull/1040

**Comment by deandreamatias** [comment]
I created a Discord server https://discord.gg/25KNPMJQf2
I have not configured anything, but we can do it after

**Comment by deandreamatias** [comment]
Hey @knaeckeKami @AlBlanc @ashim-kr-saha @vin-fandemand @criroc @furkankurt @Reprevise take a look to this too

**Comment by danvick** [comment]
This FormBuilder and its ecosystem have definitely become bigger than just me. 

I welcome the efforts of the whole Flutter community in maintaining the packages for their continuity and success. 

I'd like to commend the efforts of:
* @awhitford for adding tests and other code contributions
* @deandreamatias and @WilliamCunhaCardoso for helping with the closure of so many issues and adding structure to the whole package ecosystem
* @syntacops for creating Youtube videos explaining how to use the package and what it offers
* All the other 80+ community members who have contributed to the package by creating PRs
* Community members who have raised 1000+ issues, answered questions and made other contributions on other platforms such as Youtube, StackOverflow, Medium, etc.

**Reply by deandreamatias** [reply]
@danvick Maybe can be create a organisation (Flutter Form Builder Ecosystem) with some maintainers, to divide work.
In this way, the only blocking would be publish packages to pub.dev but this with a pipeline with secrets, will be solve

**Comment by deandreamatias** [comment]
I created a organization https://github.com/flutter-form-builder-ecosystem
Invite some people. If @danvick want, we can transfer form builder repositories to there and divide management of repositories

**Comment by AlBlanc** [comment]
Great, I am up to join the task force ;)

**Reply by WilliamCunhaCardoso** [reply]
Hey, thx for your help. If you want to, there is a server to discuss the repository.

https://discord.gg/Sea2JgcM

**Comment by ashim-kr-saha** [comment]
I will be happy to contribute.

**Reply by WilliamCunhaCardoso** [reply]
Hey, thx for your help. If you want to, there is a server to discuss the repository.

https://discord.gg/Sea2JgcM

**Comment by deandreamatias** [comment]
I waiting for Danvick feedback about organisation
I know that this changes and discussion take him time, but is for this reason and other that I want create an organisation.
I believe that with these new structure, we can organise and improve packages without so much Danvick dependence. Is better for him because we don't need disturb him all the time and better for development because has more people to manage the packages

Really sorry for this disturb @danvick  but I believe that is necessary to avoid more of them in the future

**Comment by deandreamatias** [comment]
Hi team!
Good news!
Danvick transfered all form repositories to https://github.com/flutter-form-builder-ecosystem (thanks a lot Danvick!!)
Now the next challenge is create a verified publisher to can transfer Danvick publisher to FlutterFormBuilderEcosystem verified publisher.

To [created a verified publisher](https://dart.dev/tools/pub/publishing#create-verified-publisher), we need a Google account and a domain. I don't know the best aprove in this cases to create a new verified publisher, but I already created a Google account. To domain the only way is buy one?

**Reply by WilliamCunhaCardoso** [reply]
Unfortunately I don't have experience with that. But about google account, maybe we should use his one.

**Reply by deandreamatias** [reply]
I bought a domain and created a verified publisher, can see [here](https://pub.dev/publishers/flutterformbuilderecosystem.com/packages).
Now @danvick can transfer packages from [danvick publisher](https://pub.dev/publishers/danvickmiller.dev/packages) to this new one

**Reply by WilliamCunhaCardoso** [reply]
I am aware he does not possess fully autonomy on such user/domain. I understand your motivation in those actions, but I still believe it may rely on his access. But let's see what he'll do.

**Comment by deandreamatias** [comment]
I sent some message with updates to Discord server but I will add here too.

- Separated all package to repositories. One package, one repository. I think that is better to manage changes, issues and develop in general.
- Add minimal docs and pipeline to this repositories, like issues and pr templates, update license (GNU General Public License v3.0). I created too [a file](https://github.com/flutter-form-builder-ecosystem/.github/blob/main/NEW_PACKAGE_TEMPLATE.md) with minimal thinks to have in new repositories
- Upload some new versions of some packages to test deploy pipeline (Spoiler: works nice)
- Created [an open collective](https://opencollective.com/flutter-form-builder-ecosystem) to add in future to readme packages. I think that will be nice keep a **Buy a coffee** to @danvick for a while (maybe a year?). After that and with open collective progrese, can destiny some quantity to Danvick, because him is the creator of all this. The current spent (14 USD/year) is a domain [lutterformbuilderecosystem.com](https://flutterformbuilderecosystem.com/) used to created a verified publisher
- Opened [a PR](https://github.com/flutter-form-builder-ecosystem/flutter_form_builder/pull/1088) with new readme estructure. All people can take a look if want. The idea is replicate this readme estructure to all repositories
- Created a [first general project](https://github.com/orgs/flutter-form-builder-ecosystem/projects?type=beta) to track and organize tasks by features/goals. The idea is created new projects with other things interrelated too

If has more updates, I will send here.

🚀

**Comment by deandreamatias** [comment]
Hi team!
New things that I made:

- Refactor all readme to new format
- Update license to BSD-3-clause
- Publish this new changes
- Some bugs solved 

Next steps:

- Share the organization and packages around communities to encourage to contribute with report issues, develop or organization
- Update pipelines to can update test coverage to codecov
- Update phone_number to Flutter 3.0 (https://github.com/flutter-form-builder-ecosystem/phone_number/pull/72)
-  Refactor readme of phone_number

In this moment, the organization has a better structure to receive new contributors, develop new features and resolve existing ones.

I seen new some people entering to Discord server and the stars (`1111` now) and likes of the package increased.
![image](https://user-images.githubusercontent.com/21011641/182033463-47a745a3-f607-41db-accf-111a2bb066a9.png)

I really believe that this ecosystem can maintain like the bigger and most used form package of Flutter 🚀

**Comment by deandreamatias** [comment]
Hi team!
I will share a message in several Flutter communities around the world. If someone want share in your country or local community, please feel free to do.

EN:
```markdown
Hello 👋🏽 !
Today I want to share with the Flutter community an interesting project: [Flutter Form Builder Ecosystem](https://github.com/flutter-form-builder-ecosystem).
You are probably already familiar with the name of the [flutter_form_builder](https://pub.dev/packages/flutter_form_builder) package. And yes, they are totally related. Flutter Form Builder Ecosystem is a new organization created **by** and **for** the community on Github, where you can find an ecosystem of packages and plugins designed for forms and validations.

Besides sharing this project, I would like to invite you to contribute to this ecosystem. From opening issues, creating pull request or even being part of the organization.

For those who contribute or simply use packages from the ecosystem, has [a server on Discord](https://discord.com/invite/25KNPMJQf2) where you can ask questions in general, see updates and much more.

Thank you 🚀
```

ES:
```markdown
Hola 👋🏽 !
Hoy quiero compartir con la comunidad de Flutter un proyecto interesante: [Flutter Form Builder Ecosystem](https://github.com/flutter-form-builder-ecosystem)
Probablemente ya os suene el nombre del paquete [flutter_form_builder](https://pub.dev/packages/flutter_form_builder). Y sí, están totalmente relacionados. Flutter Form Builder Ecosystem es una nueva organización creada **por** y **para** la comunidad en Github, donde podéis encontrar un ecosistema de packages y plugins pensados para formularios y validaciones.

Además de compartir este proyecto, me gustaría invitaros a contribuir con este ecosistema. Desde abrir issues, crear pull request o hasta hacer parte de la organización.

Para quien contribuya o simplemente uséis packages del ecosistema, hay [un servidor en Discord](https://discord.com/invite/25KNPMJQf2) donde podéis sacaros dudas en general, ver las actualizaciones y mucho más.

Gracias 🚀 
```

PT:
```markdown
Olá 👋🏽 !
Hoje eu quero compartilhar com a comunidade Flutter um projeto interessante: [Flutter Form Builder Ecosystem](https://github.com/flutter-form-builder-ecosystem).
Você provavelmente já conhece o nome do package [flutter_form_builder](https://pub.dev/packages/flutter_form_builder). E sim, eles são totalmente relacionados. Flutter Form Builder Ecosystem é uma nova organização criada **por** e **para** a comunidade no Github, onde você pode encontrar um ecossistema de packages e plugins projetados para formulários e validação.

Além de compartilhar este projeto, eu gostaria de convidar vocês a contribuir em este ecossistema. Desde abrir issues, criar pull request ou até mesmo fazer parte da organização.

Para aqueles que contribuem ou simplesmente usam os packages do ecossistema, tem [um servidor no Discord](https://discord.com/invite/25KNPMJQf2) onde você pode fazer perguntas em geral, ver atualizações e muito mais.

Obrigado 🚀
```

Thanks all
